ci/build.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+#
+# Environment variables:
+#
+# SOURCE_DIR: Set to the directory of the libgit2 source (optional)
+#     If not set, it will be derived relative to this script.
```

Comment:
What about the old CI scripts, shouldn't they just get removed if this is a mere refactoring? Also, what's the difference introduced by your refactoring? Sorry to be so pushy, but it's kind of hard to read as the only thing one can see in this refactoring is a new file being added.